src/main/java/duke/task/Deadline.java

```diff
@@ -0,0 +1,41 @@
+package duke.task;
+
+import java.util.Date;
+
+import duke.util.DateFormatter;
+
+/**
+ * Models a deadline task.
+ */
+public class Deadline extends Task {
+    protected Date by;
```

Comment:
Perhaps you should set the access modifier to private instead of protected. Same for the Event Class.

Reply:
Good catch, i agree :)